### PR TITLE
Column Family for ColumnTable

### LINKED ISSUE #6176: Syntax for altering compression settings

### LINKED ISSUE #10724: Column Family for ColumnTable

### PR DESCRIPTION
**Pull request by vlad-gogov**

### CODE REVIEW

**Review by nikvas0 — DISMISSED** [review]
kqp part looks ok

File: ydb/core/tx/schemeshard/olap/columns/update.cpp [hunk 1]
```diff
@@ -12,6 +12,44 @@ extern "C" {
 
 namespace NKikimr::NSchemeShard {
 
+bool TOlapColumnDiff::ParseFromRequest(const NKikimrSchemeOp::TOlapColumnDiff& columnSchema, IErrorCollector& errors) {
+    Name = columnSchema.GetName();
+    if (!!columnSchema.GetStorageId()) {
+        StorageId = columnSchema.GetStorageId();
+    }
+    if (!Name) {
+        errors.AddError("empty field name");
```
Comment (fexolm) [inline ydb/core/tx/schemeshard/olap/columns/update.cpp:21]: field "name" is empty

**Review by fexolm — DISMISSED** [review]

File: ydb/core/ydb_convert/table_description.cpp [hunk 2]
```diff
@@ -780,6 +784,38 @@ bool FillColumnDescription(NKikimrSchemeOp::TAlterColumnTable& out, const google
     return FillColumnDescriptionImpl(out, in, status, error);
 }
 
+bool FillColumnFamily(
+    const Ydb::Table::ColumnFamily& from, NKikimrSchemeOp::TFamilyDescription* to, Ydb::StatusIds::StatusCode& status, TString& error) {
+    to->SetName(from.name());
+    if (from.has_data()) {
+        status = Ydb::StatusIds::BAD_REQUEST;
+        error = "Field `DATA` is not supported for OLAP tables";
```
Comment (ijon) [inline ydb/core/ydb_convert/table_description.cpp:792]: This is the message that will be returned to the user.
So accuracy matters.

I would suggest adding column family name, the same way its done below.

File: ydb/core/kqp/host/kqp_gateway_proxy.cpp [hunk 3]
```diff
@@ -383,9 +385,59 @@ bool FillCreateTableDesc(NYql::TKikimrTableMetadataPtr metadata, NKikimrSchemeOp
...
-{
+bool FillColumnTableSchema(NKikimrSchemeOp::TColumnTableSchema& schema, const T& metadata, Ydb::StatusIds::StatusCode& code, TString& error) {
     Y_ENSURE(metadata.ColumnOrder.size() == metadata.Columns.size());
+
+    THashMap<TString, ui32> columnFamiliesByName;
+    ui32 columnFamilyId = 1;
+    for (const auto& family : metadata.ColumnFamilies) {
+        if (family.Data.Defined()) {
+            code = Ydb::StatusIds::BAD_REQUEST;
+            error = "Field `DATA` is not supported for OLAP tables";
```
Comment (ijon) [inline ydb/core/kqp/host/kqp_gateway_proxy.cpp:396]: Its worth adding column family name.

File: ydb/core/kqp/host/kqp_gateway_proxy.cpp [hunk 4]
```diff
@@ -1959,7 +2034,22 @@ class TKqpGatewayProxy : public IKikimrGateway {
 
             NKikimrSchemeOp::TColumnTableSchemaPreset* schemaPreset = storeDesc->AddSchemaPresets();
             schemaPreset->SetName("default");
-            FillColumnTableSchema(*schemaPreset->MutableSchema(), settings);
+
+            if (!settings.ColumnFamilies.empty()) {
+                IKqpGateway::TGenericResult errResult;
+                errResult.AddIssue(NYql::TIssue("TableStore is not support Column Family"));
```
Comment (ijon) [inline ydb/core/kqp/host/kqp_gateway_proxy.cpp:2040]: -> "does not support column families"

**Review by ijon — CHANGES_REQUESTED** [review]
Looks ok in general. But please check again error messages.

**Review by ijon — DISMISSED** [review]

**Review by ivanmorozov333 — DISMISSED** [review]

**Review by nikvas0 — DISMISSED** [review]

File: ydb/core/formats/arrow/serializer/native.cpp [hunk 5]
```diff
@@ -113,9 +113,8 @@ NKikimr::TConclusion<std::shared_ptr<arrow::util::Codec>> TNativeSerializer::Bui
     const int levelMin = codec->minimum_compression_level();
     const int levelMax = codec->maximum_compression_level();
     if (levelDef < levelMin || levelMax < levelDef) {
-        return TConclusionStatus::Fail(
-            TStringBuilder() << "incorrect level for codec. have to be: [" << levelMin << ":" << levelMax << "]"
-        );
+        return TConclusionStatus::Fail(TStringBuilder() << "incorrect level for codec `" << arrow::util::Codec::GetCodecAsString(cType)
+                                                        << "`. have to be: [" << levelMin << ":" << levelMax << "]");
```
Comment (zverevgeny) [inline ydb/core/formats/arrow/serializer/native.cpp:117]: minor: 
has to be ...
Но лучше expected:

**Review by zverevgeny — APPROVED** [review]

**Pull request merged**